src/strgen/strgen.cpp

```diff
@@ -269,6 +269,7 @@ struct HeaderFileWriter : HeaderWriter, FileWriter {
 		fprintf(this->fh, "/* This file is automatically generated. Do not modify */\n\n");
 		fprintf(this->fh, "#ifndef TABLE_STRINGS_H\n");
 		fprintf(this->fh, "#define TABLE_STRINGS_H\n");
+		fprintf(this->fh, "typedef uint32_t StringID;\n");
```

Comment:
Why this change? Can't find it in the description either. Feel a bit weird, in this PR?

Reply:
ah, found the answer myself, as otherwise you cannot precompile that header. But this feels like the wrong solution? Shouldn't we include some header that defines this type?

Reply:
I tried header inclusion, specifically `#include "strings_type.h"` and `#include "../strings_type.h"` (because it's as if it is in the table folder for the normal compile). But both give an error that it can't find the file because it's resolving the name relative to `<user defined path>/generated/table/strings.h`. So slapping a couple of `../`s to it would also not reliably get it to the right path.
And at that point the question is... how much effort are we going to put into something relatively brittle when just adding a single typedef works just fine.

Reply:
Aren't we also trying to replace `typedef` with `using` ?

Reply:
The manual suggest to include `strings_type.h` in the precompiled header as well, to resolve that issue.

I do not think it is advised to define `StringID` twice in our source code; that is just asking for trouble. So I agree that a `../` is not a good path, but than dismissing: so we should do a typedef like this, seems too short-sighted :) I am sure we can find something else that is elegant and sweet :)

Reply:
Yeah, as glx said, `using StringID = uint32_t;` if it is needed.

Reply:
Including `strings_type.h` works for me, but I also have the CMakeLists.txt change I suggested

Reply:
With glx's suggestion including "strings_type.h" seems to do the trick.